app/src/test/java/com/saltedge/authenticator/features/launcher/LauncherViewModelTest.kt

```diff
@@ -22,54 +22,82 @@ package com.saltedge.authenticator.features.launcher
 
 import com.saltedge.authenticator.features.main.MainActivity
 import com.saltedge.authenticator.features.onboarding.OnboardingSetupActivity
+import com.saltedge.authenticator.model.realm.RealmManagerAbs
 import com.saltedge.authenticator.model.repository.PreferenceRepositoryAbs
 import com.saltedge.authenticator.testTools.TestAppTools
 import com.saltedge.authenticator.tool.secure.PasscodeToolsAbs
-import org.junit.Assert
+import org.junit.Assert.*
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mockito.Mockito
 import org.robolectric.RobolectricTestRunner
 
 @RunWith(RobolectricTestRunner::class)
-class LauncherPresenterTest {
+class LauncherViewModelTest {
+
+    private lateinit var viewModel: LauncherViewModel
+    private val mockPreferenceRepository = Mockito.mock(PreferenceRepositoryAbs::class.java)
+    private val mockPasscodeTools = Mockito.mock(PasscodeToolsAbs::class.java)
+    private val mockRealmManager = Mockito.mock(RealmManagerAbs::class.java)
+
+    @Before
+    fun setUp() {
+        viewModel = LauncherViewModel(
+            appContext = TestAppTools.applicationContext,
+            preferenceRepository = mockPreferenceRepository,
+            passcodeTools = mockPasscodeTools,
+            realmManager = mockRealmManager
+        )
+    }
 
     @Test
     @Throws(Exception::class)
-    fun setupApplicationTest() {
-        val presenter = createPresenter()
+    fun getNextActivityClassTest() {
         Mockito.doReturn(false).`when`(mockPreferenceRepository).passcodeExist()
-        presenter.setupApplication()
 
-        Mockito.verify(mockPasscodeTools).replacePasscodeKey(TestAppTools.applicationContext)
+        assertEquals(viewModel.getNextActivityClass(), OnboardingSetupActivity::class.java)
 
         Mockito.doReturn(true).`when`(mockPreferenceRepository).passcodeExist()
-        presenter.setupApplication()
 
-        Mockito.verifyNoMoreInteractions(mockPasscodeTools)
+        assertEquals(viewModel.getNextActivityClass(), MainActivity::class.java)
     }
 
     @Test
     @Throws(Exception::class)
-    fun getNextActivityClassTest() {
-        val presenter = createPresenter()
+    fun onLifeCycleResumeTestCase1() {
```

Comment:
for complicated test cases please add description like
given:  
when:  
then: